sdcard-images-utils/nxp/patches/linux-imx/0064-drivers-media-i2c-adsd3500.c-Acknowledgement-read-da.patch

```diff
@@ -0,0 +1,89 @@
+From 78b08cdfda0ebb71c5f35ffadc111b0c2aefa61e Mon Sep 17 00:00:00 2001
+From: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
+Date: Mon, 12 Aug 2024 17:09:47 +0530
+Subject: [PATCH] drivers: media: i2c: adsd3500.c: Acknowledgement read data on
+ every header and payload data
+
+Signed-off-by: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
+---
+ drivers/media/i2c/adsd3500.c | 45 +++++++++++++++++++++++++++++++++---
+ 1 file changed, 42 insertions(+), 3 deletions(-)
+
+diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
+index 524bb0c9c51c..0e5b17c15aae 100644
+--- a/drivers/media/i2c/adsd3500.c
++++ b/drivers/media/i2c/adsd3500.c
+@@ -1311,6 +1311,30 @@ static int adsd3500_parse_dt(struct adsd3500 *priv){
+ 	return 0;
+ }
+ 
++static int adsd3500_read_ack(struct v4l2_subdev *sd){
++
++	struct adsd3500 *adsd3500 = to_adsd3500(sd);
++	struct device *dev = adsd3500->dev;
++	struct i2c_client *client = to_i2c_client(dev);
++	uint8_t read_val[4];
++	int ret;
++
++	ret = i2c_master_recv(client, read_val, sizeof(read_val));
++	if (ret < 0) {
++		dev_err(adsd3500->dev, "Read ACK cmd failed.\n");
++		return -EIO;
++	}
++	else{
++		if(read_val[0] != 0x0B){
++			dev_err(adsd3500->dev, "ACK ERROR response d0: %02X  d1: %02X d2: %02X d3: %02X\n", read_val[0], read_val[1], read_val[2], read_val[3]);
++			return -ENXIO;
++		}
++		dev_dbg(adsd3500->dev, "ACK Received\n");
++	}
++
++	return 0;
++}
++
+ static int adsd3500_send_host_boot_data(struct v4l2_subdev *sd, uint8_t* data, int size)
+ {
+ 	struct adsd3500 *adsd3500 = to_adsd3500(sd);
+@@ -1350,9 +1374,18 @@ static int adsd3500_send_host_boot_data(struct v4l2_subdev *sd, uint8_t* data, i
+ 		dev_dbg(adsd3500->dev, "Send Header data\n");
+ 		ret = i2c_master_send(client, header_packet, HEADER_SIZE_IN_BYTES);
+ 		if (ret < 0) {
+-			dev_err(adsd3500->dev, "Failed to write the header packet of segment %d\n", curr_segment);
++			dev_err(adsd3500->dev, "Failed to write the header packet of segment %d\n", curr_segment + 1);
+ 			return -EIO;
+ 		}
++		if(header_packet[03] == 0x55){
```

Comment:
* Add a space before the '(' and after the ')'.
* Add a comment on the error "!= 0x55" means.